### PR TITLE
Add some css decorations

### PR DESCRIPTION
**Pull request by andrejsharapov**
## What is the purpose of this pull request?

- [ ] New feature
- [ ] Bugfix
- [ ] Breaking change
- [x] Styles
- [ ] Documentation update
- [x] Other

## Description

I've added some dynamism to the main page and made some improvements to the code.

## Checklist

- [x] replace svg-background in Hero component to css pattern;
- [x] align to edge and add some dynamic for `LibraryCard's`;
- [x] change waves in ``HomeSection`;
- [x] other minor changes.

## Before submitting the PR, please make sure you do the following

- [x] Read the Contributing Guidelines.
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #yyy`).

### CODE REVIEW

**Comment by mririgoyen** [comment]
I'll take a look at this tonight and provide my feedback.

But may I please ask that you do not implement a git commit message style that we do not intend to follow. It'll make the history look out-of-place if some message follow some sort of style and others do not. [As I stated before](https://github.com/Pictogrammers/pictogrammers.com/issues/34#issuecomment-1407652972), we're not planning to adhere to any kind of message style for this project as the overhead does not provide us with any benefits.

P.S. I hope you don't think contributing to us is a "chore"! 🤣

File: components/Header/Header.module.scss [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
+  margin: 0 auto;
   display: flex;
+  align-items: center;
   justify-content: space-between;
-  margin: 0 auto;
-  max-height: 100px;
   max-width: $width-max;
+  max-height: 6.25rem;
   padding: 1.5rem;
+  background-color: hsl(var(--secondary-color));
```
Comment (mririgoyen) [inline components/Header/Header.module.scss]: It looks like you may not have had the stylelint plugin working when making these changes. A lot of rules are arbitrarily reordered when they should be alphabetical. Please make sure everything is passing `npm run lint` before committing. I'll get a pipeline job in place to address this shortly.

File: components/Header/Header.module.scss [hunk 2]
```diff
@@ -23,36 +24,36 @@
...
   margin-right: 1rem;
+  border: none;
 
   .wordmark {
     display: none;
   }
 
   .monogram {
-    width: 35px;
+    width: 2.1875rem;
```
Comment (mririgoyen) [inline components/Header/Header.module.scss]: Is there a reason you converted many `px` values to `rem`? There's very specific places where `px` are preferred, namely around graphic sizing which, in this example, is the case. I'd prefer to see `px` to remain in `px` unless it directly affects font-size.

File: components/Header/Header.module.scss [hunk 3]
```diff
@@ -137,29 +139,29 @@
...
+    display: none;
+    width: calc(100% + 2.75rem);
+    flex-direction: column;
+    background-color: hsl(var(--white));
+    border-radius: 0.3125rem 0 0.3125rem 0.3125rem;
+    box-shadow: var(--box-shadow);
 
     & :not(:last-child) {
-      border-bottom: 1px solid hsl(var(--light-grey));
+      border-bottom: thin solid hsl(var(--light-grey));
```
Comment (mririgoyen) [inline components/Header/Header.module.scss]: No, please use exact values.

File: components/Hero/Hero.module.scss [hunk 4]
```diff
@@ -27,9 +38,11 @@
   letter-spacing: -3px;
 }
 
-@media only screen and (max-width: $mobile-width) {
-  section:not(.full) {
-    background-position: 0 2rem;
+@media ($mobile-width >= width) {
```
Comment (mririgoyen) [inline components/Hero/Hero.module.scss]: Please leave the media queries as they were written.

File: components/Hero/Hero.tsx [hunk 5]
```diff
@@ -12,7 +12,12 @@ const ScrollButton = styled(Button)({
   fontSize: 16,
   marginBottom: '12rem',
   padding: '.5rem 1rem',
-  textTransform: 'none'
+  textTransform: 'none',
+  transition: 'letter-spacing 200ms linear',
+  
+  '&:hover': {
+    letterSpacing: 1.1,
+  },
```
Comment (mririgoyen) [inline components/Hero/Hero.tsx]: Please revert this change.

File: components/HomeSection/HomeSection.module.scss [hunk 6]
```diff
@@ -4,38 +4,52 @@
...
     z-index: 2;
+    max-width: 67.5rem;
+    margin: 0 auto 1rem;
+    padding: 0 2rem;
+    letter-spacing: -0.125rem;
+
+    font: {
+      size: 2.5rem;
+      weight: 600;
+    }
```
Comment (mririgoyen) [inline components/HomeSection/HomeSection.module.scss]: While I appreciate the ability to nest rules in this way, it makes using automatic alphabetizing more difficult and I would prefer we do not do this.

File: components/HomeSection/HomeSection.module.scss [hunk 7]
```diff
@@ -46,26 +60,28 @@
...
   &::after {
-    background-color: hsl(var(--primary-color));
     content: '';
-    height: 100px;
     position: absolute;
-    width: 100%;
     z-index: 1;
+    width: 100%;
+    height: 6.25rem;
+    background-color: hsl(var(--primary-color) / 100%);
```
Comment (mririgoyen) [inline components/HomeSection/HomeSection.module.scss]: The `100%` isn't necessary here.

File: components/HomeSection/HomeSection.module.scss [hunk 8]
```diff
@@ -46,26 +60,28 @@
...
+    transform: scale(-1, -1);
+    background-color: hsl(var(--light-grey));
   }
 }
 
-@media only screen and (max-width: $mobile-width) {
+@media (46rem >=width) {
```
Comment (mririgoyen) [inline components/HomeSection/HomeSection.module.scss]: No magic numbers please. Where is `46rem` coming from?

File: components/LibraryCard/LibraryCard.module.scss [hunk 9]
```diff
@@ -32,13 +43,14 @@
   .image {
     position: relative;
     width: 100%;
+    transition: all 0.25s linear;
```
Comment (mririgoyen) [inline components/LibraryCard/LibraryCard.module.scss]: Please remove this transition.

**Review by mririgoyen — CHANGES_REQUESTED** [review]
Generally, I think the style changes are good.

- I like the new grid, the removal of the SVG assets, and the extra wave.
- I like dropping the containing box around the library cards.

What I would like to see changed:

- Remove the animation on the hover of "Scroll to Learn More."
- Remove the all hover states on the library cards. I think they fit the style of the entire website uncontained.

Bugs:
- Please check the site in responsive sizes. There is currently a horizontal scroll. It looks likely due to the width of the buttons in the library cards.
Screenshot 2023-01-30 at 8 15 50 PM

Another Notes:
- As mentioned in the code comments, it doesn't look like the changes went through our stylelinter. Please run `npm run lint` to find errors in our rules. I'll be working on adding that to the pipeline next so we can enforce consistent code.

**Comment by andrejsharapov** [comment]
Good. Today I will make all the corrections.

A few more wishes:
- It would be nice to make a template for PR.
- Add a CONTRIBUTING file or create a Wiki so other members can follow the instructions and not make the same mistakes I did.

**Comment by mririgoyen** [comment]
> * It would be nice to make a template for PR.

I agree, I will be creating these soon. Same for issues.

> * Add a CONTRIBUTING file or create a Wiki so other members can follow the instructions and not make the same mistakes I did.

Our contributing information is linked in the README: https://pictogrammers.com/docs/contribute/website/.

**Comment by andrejsharapov** [comment]
> Our contributing information is linked in the README: [pictogrammers.com/docs/contribute/website](https://pictogrammers.com/docs/contribute/website/).

I meant something like `Code style` mentioning stylelint.

**Comment by mririgoyen** [comment]
> > Our contributing information is linked in the README: [pictogrammers.com/docs/contribute/website](https://pictogrammers.com/docs/contribute/website/).
> 
> I meant something like `Code style` mentioning stylelint.

Yep, the documentation needs to be expanded on. There's [already an issue open for it](https://github.com/Pictogrammers/pictogrammers.com/issues/14). 😄

File: components/Header/Header.module.scss [hunk 10]
```diff
@@ -1,9 +1,10 @@
 .root {
   align-items: center;
+  background-color: hsl(var(--secondary-color));
   display: flex;
   justify-content: space-between;
   margin: 0 auto;
-  max-height: 100px;
+  max-height: 6.25rem;
```
Comment (mririgoyen) [inline components/Header/Header.module.scss]:
```suggestion
  max-height: 100px;
```

File: components/Hero/Hero.module.scss [hunk 11]
```diff
@@ -20,16 +30,19 @@
...
 
 .heading {
   font-size: clamp(1rem, 6vw, 5rem);
   letter-spacing: -3px;
 }
 
-@media only screen and (max-width: $mobile-width) {
-  section:not(.full) {
-    background-position: 0 2rem;
+@media ($mobile-width >= width) {
```
Comment (mririgoyen) [inline components/Hero/Hero.module.scss]:
```suggestion
@media only screen and (max-width: $mobile-width) {
```

File: components/HomeSection/HomeSection.module.scss [hunk 12]
```diff
@@ -10,33 +10,41 @@
...
     padding: 0 1rem 1rem;
-    position: relative;
-    z-index: 2;
   }
 }
 
 .container {
   margin: 0 auto;
-  max-width: 1080px;
+  max-width: 67.5rem;
```
Comment (mririgoyen) [inline components/HomeSection/HomeSection.module.scss]: We're dealing with viewport width here, pixels is easier to understand as that is how monitors resolutions are measured.

```suggestion
  max-width: 1080px;
```

File: components/LibraryCard/LibraryCard.tsx [hunk 13]
```diff
@@ -17,16 +17,14 @@ export interface LibraryCardProps {
...
 
 const LibraryButton = styled(Button)(({ theme }) => ({
-  '&:hover': {
-    backgroundColor: 'white',
-    boxShadow: '0 1px 1px rgba(0, 0, 0, 0.12), 0 2px 2px rgba(0, 0, 0, 0.12), 0 4px 4px rgba(0, 0, 0, 0.12), 0 8px 8px rgba(0, 0, 0, 0.12), 0 16px 16px rgba(0, 0, 0, 0.12)',
-    color: theme.palette.primary.main
-  },
   backgroundColor: 'white',
   borderBottom: 'none',
+  borderRadius: 24,
```
Comment (mririgoyen) [inline components/LibraryCard/LibraryCard.tsx]: The pill shape of the button does not match any other button's style on the site, so let's remove this.

```suggestion
```

File: styles/pages/index.module.scss [hunk 14]
```diff
@@ -21,7 +23,7 @@
     grid-gap: 3rem 5rem;
     grid-template-columns: repeat(auto-fit, minmax(280px, 1fr));
     justify-items: center;
-    max-width: 1080px;
+    max-width: 67.5rem;
```
Comment (mririgoyen) [inline styles/pages/index.module.scss]:
```suggestion
    max-width: 1080px;
```

File: styles/pages/index.module.scss [hunk 15]
```diff
@@ -57,13 +59,19 @@
...
 .libraries {
+  --col-size: 25rem;
+
   align-items: center;
   display: grid;
-  grid-gap: 2rem 3rem;
-  grid-template-columns: repeat(auto-fit, minmax(320px, 1fr));
-  justify-items: center;
+  grid-gap: 3rem;
+  grid-template-columns: repeat(auto-fit, minmax(var(--col-size), 1fr));
```
Comment (mririgoyen) [inline styles/pages/index.module.scss]: Let's not use CSS variables here and be explicit here and in the media query. I get what you're doing here, but its just not a pattern we've done in this repo.

File: components/LibraryCard/LibraryCard.module.scss [hunk 16]
```diff
@@ -26,11 +36,17 @@
...
+  @media only screen and (max-width: $mobile-width) {
+    background-color: hsl(var(--white)/12%);
+    box-shadow: var(--box-shadow);
+  }
 }
 
 .content {
   .image {
     position: relative;
+    transition: all .2s linear;
```
Comment (mririgoyen) [inline components/LibraryCard/LibraryCard.module.scss]: Still need to remove this transition.

**Review by mririgoyen — CHANGES_REQUESTED** [review]
There's still some code changes that need to happen here. Once they're done, I will pull down the branch to do some manual testing before giving it the 👍🏼. There are also merge conflicts that need to be resolved.

File: components/LibraryCard/LibraryCard.tsx (hunk 13 above)
Comment (andrejsharapov) [reply components/LibraryCard/LibraryCard.tsx]: The site navigation and the "Scroll to Learn More" button have a more circular shape.

Comment (mririgoyen) [reply components/LibraryCard/LibraryCard.tsx]: Only on hover. The header follows that pattern to match the search box. The hero does it to match the header because of proximity. Neither area really "buttons." All elements that look like buttons should follow the Material guidelines as per the MUI `<Button/>` component.

File: components/LibraryCard/LibraryCard.module.scss (hunk 16 above)
Comment (andrejsharapov) [reply components/LibraryCard/LibraryCard.module.scss]: Transition smoothes the behavior of the icon when hovering over the card.

Comment (mririgoyen) [reply components/LibraryCard/LibraryCard.module.scss]: What I'm asking is to remove all transitions from those cards. Keep the style as you have proposed before hover, but make the buttons the only clickable areas. 😄

**Comment by andrejsharapov** [comment]
Looks like it's done)

> **Note**  Several interesting observations.
1. This turned out to be a little more difficult than I thought for several reasons: I stopped using pixels in my work a long time ago and work with `rem` and/or `em` more often. Plus, you're using px and rem on the same "system":
```scss
// variables.scss
// px
$tablet-width: 910px;
$mobile-width: 736px;
// but rem
$width-max: 96rem;
```
2. And yet, the choice of such breakpoints is not entirely clear to me, they are very different from the generally accepted ones (768, 992, 1024, 1200, etc.).
3. In some places it is preferable to use variables than to duplicate code with different values.
4. Many places use questionable pixels like 5/30/33/175 and others. It would be nice to translate them to 4/8/32...

In general, I liked the work. I'll suggest more improvements over time if you don't mind.

**Comment by mririgoyen** [comment]
Definitely, we love suggestions, feedbacks, and PRs.

My apologies about some of the inconsistencies in terms of `px`/`rem`. The site was put together by myself over the past 2 months. When there's a lot to do, sometimes details slip through the cracks. There's still a lot to do in terms of features and in terms of Developer Experience.

Let's see if I can address/justify any of your concerns and see if we can find places for improvement.

1. My philosophy on pixels vs rems is like comparing inches to centimeters. For general "good enough" cases, rems typically do fine. I tend to use rems for padding, margin, gaps, etc., anything that doesn't need to be too precise. Rems or vh/vw always get used for font-sizes for obvious reasons. When you need that precision, pixels are much easier to work with. (e.g. `1px` vs `0.0625rem`) It's less to write, it's easier, it's less bytes in the file. All that being said, there are definitely some inconsistencies that could be cleaned up following that mantra.

2. There is no hard-and-fast rule on breakpoints. There's lots of information on approximate screen resolution ranges for different devices, but when it comes down to it, breakpoints need to make sense to the website. In our case, the breakpoints are close to the top of the range for each device, but are adjusted slightly to better handle the content we're offering without having to make a ton of little one-off rules.

3. I agree here and think we can improve on this, but we should do it all at once and then remain consistent.

4. I'm not sure I understand your concern exactly, why are 5/30/33/175 questionable?

**Comment by andrejsharapov** [comment]
> I'm not sure I understand your concern exactly, why are 5/30/33/175 questionable?

Try to avoid odd numbers. You have probably noticed that many modern grids, fonts, and others use even values that are multiples of 4 (8, 12, 16, 24). The Material also uses these values. MUI used on this site is also built on this system. This will also be useful for the future margin/padding system.

This applies to point 1 as well. `px` is best used to set the size of decorative elements, e.g. `border-width`, but and here I'd better apply the `thin, medium or thick`, for everything else (margins, padding, max-/width, media query, font-size etc - `rem`. 

Indents of 12, 16, 24, 32, 48, 56px and above are easier to use and easy to manipulate:
```js
11px = 0.6875rem;
// vs
12px = 0.75rem;
```
```js
33px = 2.0625rem;
// vs
32px = 2rem;
```
or when creating variables/mixins:
```js
$base_px = 0.25rem;
$font_size = calc(#{$base_px} * 4);
```
In the future, it will become easier to translate and navigate in `em/rem`, without `0.0000625` :v:
